Stripe/STPBINRange.m

```diff
@@ -142,4 +203,56 @@ + (instancetype)mostSpecificBINRangeForNumber:(NSString *)number {
     }]];
 }
 
++ (void)retrieveBINRangesForPrefix:(NSString *)binPrefix completion:(STPRetrieveBINRangesCompletionBlock)completion {
+    static NSMutableDictionary<NSString *, NSArray<STPRetrieveBINRangesCompletionBlock> *> *sPendingRequests = nil;
+    static NSMutableDictionary<NSString *, NSArray<STPBINRange *> *> *sRetrievedRanges = nil;
+    static dispatch_queue_t sRetrievalQueue = nil;
+    
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sRetrievalQueue = dispatch_queue_create("com.stripe.retrieveBINRangesForPrefix", DISPATCH_QUEUE_SERIAL);
+        sPendingRequests = [NSMutableDictionary new];
+        sRetrievedRanges = [NSMutableDictionary new];
+    });
+    
+    dispatch_async(sRetrievalQueue, ^{
+        NSString *binPrefixKey = [binPrefix stp_safeSubstringToIndex:6];
```

Comment:
can we comment what 6 signifies here? Maybe make it a constant too, since it's used twice?

Reply:
updated in second patch